renderdoc/driver/gl/wrappers/gl_texture_funcs.cpp

```diff
@@ -1372,22 +1372,63 @@ void WrappedOpenGL::glCopyImageSubData(GLuint srcName, GLenum srcTarget, GLint s
       {
         TextureData &dstData = m_Textures[dstrecord->GetResourceID()];
 
-        if(srcX == 0 || srcY == 0 || srcZ == 0 || dstX == 0 || dstY == 0 || dstZ == 0)
+        GLsizei srcLevelWidth = RDCMAX(1, srcData.width >> srcLevel);
+        GLsizei srcLevelHeight =
+            srcData.dimension >= 2 ? RDCMAX(1, srcData.height >> srcLevel) : srcData.height;
+        GLsizei srcLevelDepth =
+            srcData.dimension >= 3 ? RDCMAX(1, srcData.depth >> srcLevel) : srcData.depth;
```

Comment:
Unfortunately the `dimension` stored here is still following the weird GL concept where an array is considered 3D. You would need to check `curType` to see if it's an array texture or 3D, etc.